tests/python_functional/src/syslog_ng_config/statements/destinations/file_destination.py

```diff
@@ -25,21 +25,23 @@
 import src.testcase_parameters.testcase_parameters as tc_parameters
 from src.driver_io.file.file_io import FileIO
 from src.syslog_ng_config.statements.destinations.destination_driver import DestinationDriver
-from src.syslog_ng_config.statements.destinations.destination_reader import DestinationReader
 
 
 class FileDestination(DestinationDriver):
     def __init__(self, file_name, **options):
         self.driver_name = "file"
         self.path = Path(tc_parameters.WORKING_DIR, file_name)
+        self.io = FileIO(self.path)
         super(FileDestination, self).__init__([self.path], options)
-        self.destination_reader = DestinationReader(FileIO)
 
     def get_path(self):
         return self.path
 
     def read_log(self):
-        return self.destination_reader.read_logs(self.get_path(), counter=1)[0]
+        return self.read_logs(1)[0]
 
     def read_logs(self, counter):
-        return self.destination_reader.read_logs(self.get_path(), counter=counter)
+        return self.io.read_number_of_lines(counter)
```

Comment:
`MessageReader` meant to be an abstraction that could have been able to read/buffer data from a stream-like medium and extract messages from it based on a configurable separator. What about multi-line and framed messages now?

Reply:
Thanks, for the very good questions. 
Not yet think about this, but I can imagine two options how we can handle multi-line case for file drivers:
1, strict way: in this case we will implement such an API which matches with syslog-ng settings: prefix, garbage and so on. This could work, but for the first thought it can be overhead.
2, loose way: in this case we just read everything from the destination side into one string object and then we can assert for regexp matching in it.

In case of framed messages the low level IO class (which is not yet exist) should handle the working mode.

Please correct me if I misunderstood something.

Reply:
Message reader was a good abstraction that can handle everything you mentioned in one place, in a configurable manner.
The implementation was unnecessarily complex, but the interface could have been really good if it had been used in the right place.

Anyway, you can add this or something similar back when you need it while implementing multi-line/framed message support. (The regexp trickery in each multiline test is not a clean way of writing tests in my opinion.)

Reply:
Yes, I would like to add this part later, when it is actual by testcases. I will fix soon your other review comment.